src/storage/src/source/mysql/replication/events.rs

```diff
@@ -177,3 +177,105 @@ pub(super) async fn handle_query_event(
 
     Ok(())
 }
+
+/// Handles RowsEvents from the MySQL replication stream. These events contain
+/// insert/update/delete events for a single transaction or committed statement.
+///
+/// We use these events to update the dataflow with the new rows, and return a new
+/// frontier with which to advance the dataflow's progress.
+pub(super) async fn handle_rows_event(
+    event: RowsEventData<'_>,
+    config: &RawSourceCreationConfig,
+    stream: &BinlogStream,
+    table_id_map: &mut BTreeMap<u64, UnresolvedItemName>,
+    skipped_table_ids: &mut BTreeSet<u64>,
+    errored_tables: &BTreeSet<UnresolvedItemName>,
+    table_info: &BTreeMap<UnresolvedItemName, (usize, MySqlTableDesc)>,
+    new_gtid: &GtidPartition,
+    rewinds: &BTreeMap<UnresolvedItemName, ([Capability<GtidPartition>; 2], RewindRequest)>,
+    data_output: &mut AsyncOutputHandle<
+        GtidPartition,
+        Vec<((usize, Result<Row, DefiniteError>), GtidPartition, i64)>,
+        TeeCore<GtidPartition, Vec<((usize, Result<Row, DefiniteError>), GtidPartition, i64)>>,
+    >,
+    data_cap_set: &CapabilitySet<GtidPartition>,
```

Comment:
is it time to create a struct holding onto all this context?

Reply:
I was wondering the same thing. I'll see if I can refactor this to make it cleaner!